lib/class-wp-rest-category-search-handler.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * REST API: WP_REST_Category_Search_Handler class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 5.5.0
+ */
+
+/**
+ * Core class representing a search handler for categories in the REST API.
+ *
+ * @since 5.5.0
+ *
+ * @see WP_REST_Search_Handler
+ */
+class WP_REST_Category_Search_Handler extends WP_REST_Search_Handler {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 5.0.0
+	 */
+	public function __construct() {
+		$this->type = 'category';
+	}
+
+	/**
+	 * Searches the object type content for a given search request.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @param WP_REST_Request $request Full REST request.
+	 * @return array Associative array containing an `WP_REST_Search_Handler::RESULT_IDS` containing
+	 *               an array of found IDs and `WP_REST_Search_Handler::RESULT_TOTAL` containing the
+	 *               total count for the matching search results.
+	 */
+	public function search_items( WP_REST_Request $request ) {
+
+		$categories = get_categories( array( 'get' => 'all' ) );
+
+		if ( ! empty( $request['search'] ) ) {
+			$category_search = $request['search'];
+		}
+
+		$category_search = apply_filters( 'rest_category_search_query', $category_search, $request );
+
+		$found_ids = [];
+		foreach ( $categories as $category ) {
+			$name_match = stripos( $category->name, $category_search ) !== false;
```

Comment:
Is there any way to include this in the actual query instead of matching in PHP?

Reply:
[Apparently](https://developer.wordpress.org/reference/classes/wp_term_query/__construct/) It is!